src/Worker.Extensions.DurableTask/DurableHttpResponseConverter.cs

```diff
@@ -0,0 +1,138 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Text;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.Extensions.Primitives;
+
+namespace Microsoft.Azure.Functions.Worker;
+internal class DurableHttpResponseConverter : JsonConverter<DurableHttpResponse>
```

Comment:
Can you explain why a full converter was necessary here? Particularly, why did `JsonPropertyName` attributes not work? I know I mentioned writing your own converter, which may be what needs to happen, but I misspoke and really meant to just customize JSON conversion (through attributes or converter).

I'd recommend using attributes if possible, as they are less code to maintain. You can start with an attribute approach, serialize both the in-proc and out-of-proc types using the respective JSON libraries, and compare them for what you need to change.

Reply:
I wrote the full converter based off of your previous comment, but without it I kept running into serialization errors like `Unexpected token while deserializing object: PropertyName. Path '[0].uri', line 1, position 36.` when deserializing the JSON.

Reply:
Lets sync tomorrow on this - I was able to get conversion working with just attributes.